### PR TITLE
OC-595 Log information about messages on Kafka topics

### PR DESCRIPTION
**Pull request by bvdzwet**
Adds KafkaLogger for use with consumer configuration on Kafka topics to
be able to log data and metadata of messages that are posted on a topic.

Adds osgp-adapter-kafka-logging as deployable project in servlet
containers with configuration of consumers for currently known topics
in the GXF code base.

The configurations of the different logging consumers are subject to a
Spring conditional, following a boolean property (for instance
peakshaving.kafka.logging.enabled). Default this enabled property is set
to false, which should prevent warning or error logging if a Kafka
broker is not available for certain topics in a monitored environment.

The consumed messages for the configured topics will be logged to a file
specified in the logback configuration with an appender named MESSAGES.
This appender leaves out most of the default stuff that is normally
added as context with logged data, leaving only the date and time of
logging, and the message that is logged.

### CODE REVIEW

File: osgp/shared/osgp-kafka-logger/src/main/java/org/opensmartgridplatform/kafka/logging/CountDownKafkaLogger.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
```
Comment (robindenadel) [inline osgp/shared/osgp-kafka-logger/src/main/java/org/opensmartgridplatform/kafka/logging/CountDownKafkaLogger.java:1]: if we would like to use this class in the existing consumer test in closed source, I think it should be moved to the project shared.

File: osgp/platform/osgp-adapter-kafka-logging/src/main/java/org/opensmartgridplatform/adapter/kafka/logging/config/KafkaDistributionAutomationConsumerConfig.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/**
```
[the remark this answers is not in the document]
Comment (bvdzwet) [reply osgp/platform/osgp-adapter-kafka-logging/src/main/java/org/opensmartgridplatform/adapter/kafka/logging/config/KafkaDistributionAutomationConsumerConfig.java:1]: removed

File: osgp/platform/osgp-adapter-kafka-logging/src/main/java/org/opensmartgridplatform/adapter/kafka/logging/config/DistributionAutomationLoggingEnabled.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/**
```
[the remark this answers is not in the document]
Comment (bvdzwet) [reply osgp/platform/osgp-adapter-kafka-logging/src/main/java/org/opensmartgridplatform/adapter/kafka/logging/config/DistributionAutomationLoggingEnabled.java:1]: removed

File: osgp/platform/osgp-adapter-kafka-logging/src/main/java/org/opensmartgridplatform/adapter/kafka/logging/infra/kafka/in/MeterReadingConsumer.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/**
```
[the remark this answers is not in the document]
Comment (bvdzwet) [reply osgp/platform/osgp-adapter-kafka-logging/src/main/java/org/opensmartgridplatform/adapter/kafka/logging/infra/kafka/in/MeterReadingConsumer.java:1]: removed

File: osgp/platform/osgp-adapter-kafka-logging/src/main/resources/osgp-adapter-kafka-logging.properties [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+distributionautomation.kafka.topic=g_osgp_distributionautomation_monitoring_dev
+
+distributionautomation.kafka.consumer.key.deserializer=org.apache.kafka.common.serialization.StringDeserializer
+distributionautomation.kafka.consumer.value.deserializer=org.opensmartgridplatform.adapter.kafka.da.serialization.MeterReadingDeserializer
+distributionautomation.kafka.consumer.group.id=DistributionAutomationLoggingConsumer
+distributionautomation.kafka.consumer.enable.auto.commit=true
+distributionautomation.kafka.consumer.poll.timeout=3000
+distributionautomation.kafka.consumer.concurrency=1
+distributionautomation.kafka.consumer.auto.offset.reset=latest
+
```
[the remark this answers is not in the document]
Comment (bvdzwet) [reply osgp/platform/osgp-adapter-kafka-logging/src/main/resources/osgp-adapter-kafka-logging.properties:18]: removed

File: osgp/platform/osgp-adapter-kafka-logging/src/test/java/org/opensmartgridplatform/adapter/kafka/logging/infra/kafka/in/MeterReadingConsumerTest.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
+/**
```
[the remark this answers is not in the document]
Comment (bvdzwet) [reply osgp/platform/osgp-adapter-kafka-logging/src/test/java/org/opensmartgridplatform/adapter/kafka/logging/infra/kafka/in/MeterReadingConsumerTest.java:1]: removed

File: osgp/platform/osgp-adapter-kafka-logging/src/test/resources/osgp-adapter-kafka-logging-test-distributionautomation.properties [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# =========================================================
```
[the remark this answers is not in the document]
Comment (bvdzwet) [reply osgp/platform/osgp-adapter-kafka-logging/src/test/resources/osgp-adapter-kafka-logging-test-distributionautomation.properties:1]: removed

**Pull request merged**